doc/content/design/sm-supported-image-formats.md

```diff
@@ -22,32 +22,39 @@ available formats.
 # Design Proposal
 
 To expose the available image formats to clients (e.g., XenCenter, XenOrchestra, etc.),
-we propose adding a new field called `supported-image-formats` to the Storage Manager (SM)
-module. This field will be included in the output of the `SM.get_all_records` call.
+we propose adding a new field called `supported_image_formats` to the Storage Manager
+(SM) module. This field will be included in the output of the `SM.get_all_records` call.
 
-The `supported-image-formats` field will be populated by retrieving information
+The `supported_image_formats` field will be populated by retrieving information
 from the SMAPI drivers. Specifically, each driver will update its `DRIVER_INFO`
 dictionary with a new key, `supported_image_formats`, which will contain a list
 of strings representing the supported image formats
 (for example: `["vhd", "raw", "qcow2"]`).
 
-The list designates the driver's preferred VDI format as its first entry. That
-means that when migrating a VDI, the destination storage repository will
-attempt to create a VDI in this preferred format. If the default format cannot
-be used (e.g., due to size limitations), an error will be generated.
+If a driver does not provide this information (as is currently the case with
+existing drivers), the default value will be an empty array. This signifies that
+the driver determines which format to use when creating or migrating a VDI.
```

Comment:
On migration, which of the sides decides what format to use? The receiving end? Or is there a negotiation between both sides needed?

Reply:
As I see it, if no extra information is provided, it is up to the receiving end to decide the default format. Since the supported formats are not exposed, the client cannot make any assumptions. Therefore, only the receiving side can determine its default format, and there is no negotiation between both sides.

Reply:
The other solution is to check in XAPI that the same format is available for both side and if it is the case keep the same format. Otherwise we use the destination format. It is not clear yet how to check that in XAPI so that's why using the default format of the destination is maybe easier as a first approach...

Reply:
Can the fact that it's the receiving side that decides be written here?

Reply:
Yes sure. I'm not sure which solution is better. So I'm waiting for other opinions if any and will update the PR accordingly :+1: